sdks/v4-sdk/src/utils/v4Planner.ts

```diff
@@ -117,23 +123,41 @@ export class V4Planner {
   }
 
   addTrade(trade: Trade<Currency, Currency, TradeType>, slippageTolerance?: Percent): void {
-    const actionType = trade.tradeType === TradeType.EXACT_INPUT ? Actions.SWAP_EXACT_IN : Actions.SWAP_EXACT_OUT
+    const exactOutput = trade.tradeType === TradeType.EXACT_OUTPUT
+
+    if (exactOutput) invariant(!!slippageTolerance, 'ExactOut requires slippageTolerance')
```

Comment:
how come we only check this for exactOuput? just curious

Reply:
good question I can leave a comment. on exactInput sometimes we do agg slippage checks so we don't need to know slippage, but we enver do that on exactOutput